packages/manager/modules/web-paas/src/details/service/add-addon/template.html

```diff
@@ -0,0 +1,46 @@
+<form
+    id="addAddon"
+    name="addAddon"
+    data-ng-submit="$ctrl.addon.quantity > 0 && $ctrl.orderAddon()"
+    novalidate
+>
+    <oui-modal
+        data-primary-label="{{ ::'web_paas_common_confirm' | translate }}"
+        data-secondary-action="$ctrl.goBack()"
+        data-secondary-label="{{ ::'web_paas_common_cancel' | translate }}"
+        data-on-dismiss="$ctrl.goBack()"
+        data-primary-disabled="$ctrl.disableSubmit"
+        data-heading="{{ ::'web_paas_service_add_addon_title_' + $ctrl.addonType.split('-').pop() | translate }}"
+    >
+        <p data-ng-bind="$ctrl.description"></p>
+        <web-paas-additional-option
+            disabled="$ctrl.disableNumeric"
+            option="$ctrl.addon"
+            plan="$ctrl.project.selectedPlan"
+            project="$ctrl.project"
+            on-change="$ctrl.onChange(modelValue)"
+            user="$ctrl.user"
+        >
+        </web-paas-additional-option>
+        <p>
+            <oui-spinner
+                data-size="s"
+                data-ng-if="$ctrl.disableNumeric"
+            ></oui-spinner>
+            <span
+                data-ng-if="$ctrl.disableNumeric"
+                data-translate="web_paas_service_add_addon_fetching_price"
+            ></span>
+            <span
+                data-ng-if="!$ctrl.disableNumeric"
+                data-translate="{{:: 'web_paas_service_add_addon_prorata_price_' + $ctrl.addonType.split('-').pop() }}"
```

Comment:
You can move this logic to addonType class. Create a class and have a method like getAddonTypexxx(). I have also noticed that you are using this, $ctrl.addonType.split('-'), in multiple places. It would be better to create a class.